### ISSUE TITLE
move tokenizer and serialiser tests over to generating test objects from the JSON files

### ISSUE THREAD

**Issue by willkg** [html5lib/html5lib-python#367]
From #182.

The serializer tests seem to be in `html5lib/tests/test_serializer.py`.

**Comment by willkg** [comment]
@gsnedders The tokenizer tests look like they're in `html5lib/tests/testdata/tokenizer/` already. Are there other tests you wanted to move over?

**Comment by gsnedders** [comment]
Yeah, tokenisers have been done. Basically everything using the `testdata` files should be using the same structure with subclasses of `pytest.Item`.

**Comment by willkg** [comment]
Adjusted the title accordingly. Thanks, @gsnedders !
